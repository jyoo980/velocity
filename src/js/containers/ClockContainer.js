import NameForm from "../components/NameForm";
import Time from "../components/Time";
import UnderClockMessage from "../components/UnderClockMessage";
import { determineMessageUnderClock, getCurrentTime } from "../utils/utils";
import { h } from "preact";
import { useEffect, useState } from "preact/hooks";

function ClockContainer() {
  const timePerTick = 500;
  let [currentHour, currentMinute] = getCurrentTime();

  const [minute, setMinute] = useState(currentMinute);
  const [hour, setHour] = useState(currentHour);
  const [messageUnderClock, setMessageUnderClock] = useState(
    determineMessageUnderClock(hour)
  );

  useEffect(() => {
    setMessageUnderClock(determineMessageUnderClock(hour));
  }, [minute]);

  setInterval(() => setMinute(currentMinute), timePerTick);
  setInterval(() => setHour(currentHour), timePerTick);
  console.log("ran");
  return (
    <div id="clock-wrapper">
      <NameForm />
      <Time hour={hour} minute={minute} />
      <UnderClockMessage message={messageUnderClock} />
    </div>
  );
}

export default ClockContainer;
