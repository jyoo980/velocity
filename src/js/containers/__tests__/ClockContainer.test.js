import ClockContainer from "../ClockContainer";
import { h } from "preact";
import { mount } from "enzyme";

jest.mock("../../utils/utils", () => ({
  getCurrentTime: jest.fn(() => ["9", "41"]),
  determineMessageUnderClock: jest.fn(() => "Good evening, Thog."),
}));

describe("ClockContainer", () => {
  test("ClockContainer", () => {
    const wrapper = mount(<ClockContainer />);

    expect(wrapper).toMatchSnapshot();
  });
});
