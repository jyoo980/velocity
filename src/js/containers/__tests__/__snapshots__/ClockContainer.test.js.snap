// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClockContainer updates clock and message when time changes 1`] = `
<ClockContainer>
  <div
    id="clock-wrapper"
  >
    <NameForm
      visible={true}
    />
    <Time
      hour="18"
      minute="59"
      visible={false}
    >
      <div
        id="clock"
      >
        <span
          id="hour"
        >
          18
        </span>
        <span
          id="minute"
        >
          59
        </span>
      </div>
    </Time>
    <UnderClockMessage
      message="Message Under Clock"
      visible={false}
    >
      <div
        id="message-under-clock"
      >
        Message Under Clock
      </div>
    </UnderClockMessage>
  </div>
</ClockContainer>
`;

exports[`ClockContainer updates clock and message when time changes 2`] = `
<ClockContainer>
  <div
    id="clock-wrapper"
  >
    <NameForm
      visible={true}
    />
    <Time
      hour="19"
      minute="59"
      visible={false}
    >
      <div
        id="clock"
      >
        <span
          id="hour"
        >
          19
        </span>
        <span
          id="minute"
        >
          59
        </span>
      </div>
    </Time>
    <UnderClockMessage
      message="Message Under Clock"
      visible={false}
    >
      <div
        id="message-under-clock"
      >
        Message Under Clock
      </div>
    </UnderClockMessage>
  </div>
</ClockContainer>
`;
